pkg/devfile/components.go

```diff
@@ -0,0 +1,265 @@
+	devfileData, err := data.NewDevfileData(string(data.APISchemaVersion200))
+	if err != nil {
+		t.Error(err)
+	}
+	for _, component := range inlined {
+		err = devfileData.AddComponents([]v1alpha2.Component{{
+			Name: component.Name,
+			ComponentUnion: devfilev1.ComponentUnion{
+				Kubernetes: &devfilev1.KubernetesComponent{
+					K8sLikeComponent: devfilev1.K8sLikeComponent{
+						BaseComponent: devfilev1.BaseComponent{},
+						K8sLikeComponentLocation: devfilev1.K8sLikeComponentLocation{
+							Inlined: component.Inlined,
+						},
+					},
+				},
+			},
+		},
+		})
+		if err != nil {
+			t.Error(err)
+		}
+	}
+	for _, component := range uriComp {
+		err = devfileData.AddComponents([]v1alpha2.Component{{
+			Name: component.Name,
+			ComponentUnion: devfilev1.ComponentUnion{
+				Kubernetes: &devfilev1.KubernetesComponent{
+					K8sLikeComponent: devfilev1.K8sLikeComponent{
+						BaseComponent: devfilev1.BaseComponent{},
+						K8sLikeComponentLocation: devfilev1.K8sLikeComponentLocation{
+							Uri: component.URI,
+						},
+					},
+				},
+			},
+		},
+		})
+		if err != nil {
+			t.Error(err)
+		}
+	}
+	return devfileData
+}
+
+// GetComponentsToPush returns the list of Kubernetes components to push,
+// by getting the list of Kubernetes components and removing the ones
+// referenced from a command in the devfile
+func GetKubernetesComponentsToPush(devfileObj parser.DevfileObj) ([]devfilev1.Component, error) {
+	k8sComponents, err := devfileObj.Data.GetComponents(parsercommon.DevfileOptions{
+		ComponentOptions: parsercommon.ComponentOptions{ComponentType: devfilev1.KubernetesComponentType},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	componentsMap := map[string]devfilev1.Component{}
+	for _, component := range k8sComponents {
+		componentsMap[component.Name] = component
+	}
+
+	commands, err := devfileObj.Data.GetCommands(parsercommon.DevfileOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, command := range commands {
+		componentName := ""
+		if command.Exec != nil {
+			componentName = command.Exec.Component
+		} else if command.Apply != nil {
+			componentName = command.Apply.Component
+		}
+		if componentName == "" {
+			continue
+		}
+		delete(componentsMap, componentName)
+	}
+
+	k8sComponents = make([]devfilev1.Component, len(componentsMap))
+	i := 0
+	for _, v := range componentsMap {
+		k8sComponents[i] = v
+		i++
+	}
+
+	return k8sComponents, err
+}
+
+// IsComponentDefined checks if a component with the given name is defined in a DevFile
+func IsComponentDefined(name string, devfileObj parser.DevfileObj) (bool, error) {
+	components, err := devfileObj.Data.GetComponents(common.DevfileOptions{})
+	if err != nil {
+		return false, err
+	}
+	for _, c := range components {
+		if c.Name == name {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// AddKubernetesComponentToDevfile adds service definition to devfile as an inlined Kubernetes component
```

Comment:
It's no longer just service definition. You moved the code but forgot to update the comment.